app/scripts/keyring-controller.js

```diff
@@ -173,10 +173,15 @@ module.exports = class KeyringController extends EventEmitter {
     })
   }
 
-  placeSeedWords () {
+  placeSeedWords (cb) {
     const firstKeyring = this.keyrings[0]
     const seedWords = firstKeyring.serialize().mnemonic
     this.configManager.setSeedWords(seedWords)
+
+    if (cb) {
+      cb()
+      this.emit('update')
```

Comment:
Could you explain why? It seems like this is just adding update spam.  We already transmit an update in the synchronous usage of this method.